packages/engine/Source/Shaders/Voxels/IntersectEllipsoid.glsl

```diff
@@ -26,133 +24,200 @@
 #if defined(ELLIPSOID_HAS_RENDER_BOUNDS_LONGITUDE)
     uniform vec2 u_ellipsoidRenderLongitudeMinMax;
 #endif
-#if defined(ELLIPSOID_HAS_RENDER_BOUNDS_LATITUDE_MIN_UNDER_HALF) || defined(ELLIPSOID_HAS_RENDER_BOUNDS_LATITUDE_MIN_OVER_HALF) || defined(ELLIPSOID_HAS_RENDER_BOUNDS_LATITUDE_MAX_UNDER_HALF) || defined(ELLIPSOID_HAS_RENDER_BOUNDS_LATITUDE_MAX_OVER_HALF)
-    uniform vec2 u_ellipsoidRenderLatitudeCosSqrHalfMinMax;
-#endif
-#if defined(ELLIPSOID_HAS_RENDER_BOUNDS_HEIGHT_MAX)
-    uniform float u_ellipsoidInverseOuterScaleUv;
-#endif
-#if defined(ELLIPSOID_HAS_RENDER_BOUNDS_HEIGHT_MIN)
-    uniform float u_ellipsoidInverseInnerScaleUv;
-#endif
+uniform vec2 u_ellipsoidRenderLatitudeSinMinMax;
+uniform vec2 u_clipMinMaxHeight;
 
-vec2 intersectZPlane(Ray ray)
-{
-    float o = ray.pos.z;
-    float d = ray.dir.z;
-    float t = -o / d;
-    float s = sign(o);
-
-    if (t >= 0.0 != s >= 0.0) return vec2(t, +INF_HIT);
-    else return vec2(-INF_HIT, t);
-}
+RayShapeIntersection intersectZPlane(in Ray ray, in float z) {
+    float t = -ray.pos.z / ray.dir.z;
 
-vec2 intersectUnitSphere(Ray ray)
-{
-    vec3 o = ray.pos;
-    vec3 d = ray.dir;
+    bool startsOutside = sign(ray.pos.z) == sign(z);
+    bool entry = (t >= 0.0) != startsOutside;
 
-    float b = dot(d, o);
-    float c = dot(o, o) - 1.0;
-    float det = b * b - c;
+    vec4 intersect = vec4(0.0, 0.0, z, t);
+    vec4 farSide = vec4(normalize(ray.dir), INF_HIT);
 
-    if (det < 0.0) {
-        return vec2(NO_HIT);
+    if (entry) {
+        return RayShapeIntersection(intersect, farSide);
+    } else {
+        return RayShapeIntersection(-1.0 * farSide, intersect);
     }
-
-    det = sqrt(det);
-    float t1 = -b - det;
-    float t2 = -b + det;
-    float tmin = min(t1, t2);
-    float tmax = max(t1, t2);
-
-    return vec2(tmin, tmax);
 }
 
-vec2 intersectUnitSphereUnnormalizedDirection(Ray ray)
+RayShapeIntersection intersectHeight(in Ray ray, in float relativeHeight, in bool convex)
 {
-    vec3 o = ray.pos;
-    vec3 d = ray.dir;
-
-    float a = dot(d, d);
-    float b = dot(d, o);
-    float c = dot(o, o) - 1.0;
-    float det = b * b - a * c;
-
-    if (det < 0.0) {
-        return vec2(NO_HIT);
+    // Scale the ray by the ellipsoid axes to make it a unit sphere
+    // Note: approximating ellipsoid + height as an ellipsoid
+    // Note: may be better to approximate radiiCorrection for small relative heights
```

Comment:
Does this comment indicate a future TODO, or a recommendation to the function user? If the former, it's better to put that in an issue than the code itself, where it may get overlooked.

Reply:
I removed the comment--it's irrelevant now that we aren't using this function for individual voxel intersections.